jetty-core/jetty-websocket/jetty-websocket-jetty-client/src/main/java/org/eclipse/jetty/websocket/client/ClientUpgradeRequest.java

```diff
@@ -39,28 +38,12 @@ public final class ClientUpgradeRequest implements UpgradeRequest
     private final List<ExtensionConfig> extensions = new ArrayList<>(1);
     private final List<HttpCookie> cookies = new ArrayList<>(1);
     private final Map<String, List<String>> headers = new TreeMap<>(String.CASE_INSENSITIVE_ORDER);
-    private final URI requestURI;
-    private final String host;
     private long timeout;
+    private String httpVersion;
 
     public ClientUpgradeRequest()
     {
         /* anonymous, no requestURI, upgrade request */
-        this.requestURI = null;
-        this.host = null;
-    }
-
-    /**
-     * @deprecated use {@link #ClientUpgradeRequest()} instead.
-     */
-    @Deprecated
-    public ClientUpgradeRequest(URI uri)
-    {
-        this.requestURI = uri;
-        String scheme = uri.getScheme();
-        if (!HttpScheme.WS.is(scheme) && !HttpScheme.WSS.is(scheme))
-            throw new IllegalArgumentException("URI scheme must be 'ws' or 'wss'");
-        this.host = this.requestURI.getHost();
```

Comment:
This removal feels wrong, as `WebSocketClient.connect(endPoint, URI, ClientUpgradeRequest)` should have been deprecated instead.

I would have kept `WebSocketClient.connect(endPoint, URI)`, but when I pass `ClientUpgradeRequest`, then everything is configured in it, rather than the URI separated from `ClientUpgradeRequest`, so just have 
`WebSocketClient.connect(endPoint, ClientUpgradeRequest)`.

Reply:
Opened https://github.com/jetty/jetty.project/issues/13213